### ISSUE TITLE
format bug?

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#4556]
Migrated from [rt.perl.org#7878](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=7878) (status was 'resolved')

Searchable as RT7878$

**Comment by p5pRT** [comment]
### From gimeney@gimeney.com



table\.cgi contain construction like&#8203;:
====================================================
format FORMATNAME =
@&#8203;\*
"\<tr>\<td>$r\->\{'Name'\}\</td>\</tr>"
\.\.

format TABLESTATUS = 
====================================================

It's work on my machine with ActiveState perl\.
But on Linux server it's error&#8203;:

bash$ \./table\.cgi
syntax error at \./table\.cgi line 129\, near "format TABLESTATUS "
Missing right bracket at \./table\.cgi line 164\, at end of line
Execution of \./table\.cgi aborted due to compilation errors\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=high

Site configuration information for perl v5.6.1:

Summary of my perl5 (revision 5 version 6 subversion 1) configuration:
  Platform:
    osname=MSWin32, osvers=4.0, archname=MSWin32-x86-multi-thread
    uname=''
    config_args='undef'
    hint=recommended, useposix=true, d_sigaction=undef
    usethreads=undef use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=undef d_sfio=undef uselargefiles=undef usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
  Compiler:
    cc='cl', ccflags ='-nologo -O1 -MD -DNDEBUG -DWIN32 -D_CONSOLE -DNO_STRICT -DHAVE_DES_FCRYPT  -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DPERL_MSVCRT_READFIX',
    optimize='-O1 -MD -DNDEBUG',
    cppflags='-DWIN32'
    ccversion='', gccversion='', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=undef, longlongsize=8, d_longdbl=define, longdblsize=10
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=8, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='link', ldflags ='-nologo -nodefaultlib -release  -libpath:"D:\usr\lib\CORE"  -machine:x86'
    libpth="C:\usr\lib\CORE"
    libs=  oldnames.lib kernel32.lib user32.lib gdi32.lib winspool.lib  comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib  netapi32.lib uuid.lib wsock32.lib mpr.lib winmm.lib  version.lib odbc32.lib odbccp32.lib msvcrt.lib
    perllibs=  oldnames.lib kernel32.lib user32.lib gdi32.lib winspool.lib  comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib  netapi32.lib uuid.lib wsock32.lib mpr.lib winmm.lib  version.lib odbc32.lib odbccp32.lib msvcrt.lib
    libc=msvcrt.lib, so=dll, useshrplib=yes, libperl=perl56.lib
  Dynamic Linking:
    dlsrc=dl_win32.xs, dlext=dll, d_dlsymun=undef, ccdlflags=' '
    cccdlflags=' ', lddlflags='-dll -nologo -nodefaultlib -release  -libpath:"D:\usr\lib\CORE"  -machine:x86'

Locally applied patches:
    ACTIVEPERL_LOCAL_PATCHES_ENTRY


@INC for perl v5.6.1:
    E:/usr/lib
    E:/usr/site/lib
    .


Environment for perl v5.6.1:
    HOME (unset)
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=C:\Tools\Far;E:\WINNT\system32;E:\WINNT;E:\WINNT\system32\WBEM;E:\usr\bin\;E:\Program Files\Common Files\Adaptec Shared\System;C:\WINDOWS;C:\WINDOWS\COMMAND;c:\win98\;c:\win98\command\;c:\toOLS\DN;C:\TOOLS;c:\tools\qview;c:\tools\tasm\bin;
    PERL_BADLANG (unset)
    SHELL (unset)

```
</details>

**Comment by p5pRT** [comment]
### From @tux

Are you sure that the Linux host that generates this error is also 5\.6\.1 or
are you reporting the bug from another machine\.

This problem is solved in Change 6340 by jhi@&#8203;alpha on 2000/07/11 15&#8203;:55&#8203;:47\,
which is included in all 5\.7\.x and in 5\.6\.1

the problem occurs if there is a closing brace inside double quotes in format
parameter lists\. A workaround would be

format FORMATNAME =
@&#8203;\*
"\<tr>\<td>"\.$r\->\{Name\}\."\</td>\</tr>"
\.

which should parse correct\.

\-\- 
H\.Merijn Brand    Amsterdam Perl Mongers \(http&#8203;://www\.amsterdam\.pm\.org/\)
using perl\-5\.6\.1\, 5\.7\.2 & 629 on HP\-UX 10\.20 & 11\.00\, AIX 4\.2\, AIX 4\.3\,
&nbsp;  WinNT 4\, Win2K pro & WinCE 2\.11\.  Smoking perl CORE&#8203;: smokers@&#8203;perl\.org
http&#8203;://archives\.develooper\.com/daily\-build@&#8203;perl\.org/   perl\-qa@&#8203;perl\.org
send smoke reports to&#8203;: smokers\-reports@&#8203;perl\.org\, QA&#8203;: http&#8203;://qa\.perl\.org

**Comment by p5pRT** [comment]
### From @tux

> \[gimeney@&#8203;gimeney\.com \- Tue Nov 06 12&#8203;:18&#8203;:02 2001\]&#8203;:
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> table\.cgi contain construction like&#8203;:
> ====================================================
> format FORMATNAME =
> @&#8203;\*
> "\<tr>\<td>$r\->\{'Name'\}\</td>\</tr>"
> \.

Indeed solved&#8203;:

a5&#8203;:/u/usr/merijn 102 > cat xx\.pl
\#\!/pro/bin/perl

use strict;
use warnings;

$~ = "FORMATNAME";
my $r = \{ Name => "Test" \};
write;

format FORMATNAME =
@&#8203;\*
"\<tr>\<td>$r\->\{'Name'\}\</td>\</tr>"
\.
a5&#8203;:/u/usr/merijn 103 > perl5\.6\.1 xx\.pl
\<tr>\<td>Test\</td>\</tr>
a5&#8203;:/u/usr/merijn 104 > perl5\.8\.0 xx\.pl
\<tr>\<td>Test\</td>\</tr>
a5&#8203;:/u/usr/merijn 105 > perl5\.9\.0 xx\.pl
\<tr>\<td>Test\</td>\</tr>
a5&#8203;:/u/usr/merijn 106 >

Closed

Enjoy\, Have FUN\! H\.Merijn

**Comment by p5pRT** [comment]
@tux - Status changed from 'open' to 'resolved'
